snuba/subscriptions/data.py

```diff
@@ -181,16 +189,23 @@ def from_dict(
             resolution_sec=int(data["resolution"]),
             query=data["query"],
             entity_subscription=entity_subscription,
+            metadata=metadata,
         )
 
     def to_dict(self) -> Mapping[str, Any]:
-        return {
+        subscription_data_dict = {
             "project_id": self.project_id,
             "time_window": self.time_window_sec,
             "resolution": self.resolution_sec,
             "query": self.query,
-            **self.entity_subscription.to_dict(),
         }
+        if self.entity_subscription.processors:
+            for processor in self.entity_subscription.processors:
+                subscription_data_dict.update(processor.to_dict(self.metadata))
+        if self.entity_subscription.validators:
+            for validator in self.entity_subscription.validators:
+                subscription_data_dict.update(validator.to_dict())
```

Comment:
I think the validator should also be passed the metadata. I know we don't use it currently but that will probably change.